### PR TITLE
sway: integrate with systemd

### PR DESCRIPTION
**Pull request by laalsaas**
see: https://github.com/swaywm/sway/wiki/Systemd-integration

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/programs/wayland/sway.nix [hunk 1]
```diff
@@ -123,12 +123,23 @@ in
             # Import the most important environment variables into the D-Bus and systemd
             # user environments (e.g. required for screen sharing and Pinentry prompts):
             exec dbus-update-activation-environment --systemd DISPLAY WAYLAND_DISPLAY SWAYSOCK XDG_CURRENT_DESKTOP
```
Comment (adamdicarlo0) [inline nixos/modules/programs/wayland/sway.nix:125]: this line passes `--systemd`, described as "Also update the 'systemd --user' environment if possible."

So I'm wondering, is the `systemctl --user import-environment` line necessary?

And, a bit off-topic, shouldn't the `exec dbus-update-activation-environment ...` line be an `exec_always`?

(When I rebuild my home-manager config, certain services I have that need WAYLAND_DISPLAY seem to be unable to restart, and I've found--after noticing this PR, thanks!--that manually running `systemctl --user import-environment DISPLAY WAYLAND_DISPLAY SWAYSOCK;` then enables the services to restart successfully.)

**Comment by Synthetica9** [comment]
@ofborg test sway

File: nixos/modules/programs/wayland/sway.nix (hunk 1 above)
Comment (laalsaas) [reply nixos/modules/programs/wayland/sway.nix:125]: > (When I rebuild my home-manager config, certain services I have that need WAYLAND_DISPLAY seem to be unable to restart, and I've found--after noticing this PR, thanks!--that manually running systemctl --user import-environment DISPLAY WAYLAND_DISPLAY SWAYSOCK; then enables the services to restart successfully.)

Doesn't this indicate that it's still necessary to run `systemctl --user import-environment`? It says "if possible" but who knows what "possible" means.

Comment (adamdicarlo0) [reply nixos/modules/programs/wayland/sway.nix:125]: Interesting point on the "if possible." OK, I agree, it sounds safer to leave it in!

**Review by leona-ya — APPROVED** [review]
Oh! That's exactly what I searched for. Can you rebase please, and then I'll merge :)

**Comment by adamdicarlo0** [comment]
thank you @laalsaas & @leona-ya! 🙌🏻

**Pull request merged**